monkey/infection_monkey/post_breach/actions/communicate_as_new_user.py

```diff
@@ -0,0 +1,117 @@
+import os
+import random
+import string
+import subprocess
+
+from common.data.post_breach_consts import POST_BREACH_COMMUNICATE_AS_NEW_USER
+from infection_monkey.post_breach.actions.add_user import BackdoorUser
+from infection_monkey.post_breach.pba import PBA
+from infection_monkey.telemetry.post_breach_telem import PostBreachTelem
+from infection_monkey.utils import is_windows_os
+
+CREATED_PROCESS_AS_USER_WINDOWS_FORMAT = "Created process '{}' as user '{}'."
+CREATED_PROCESS_AS_USER_LINUX_FORMAT = "Created process '{}' as user '{}'. Some of the output was '{}'."
+
+USERNAME = "somenewuser"
+PASSWORD = "N3WPa55W0rD!1"
+
+logger = logging.getLogger(__name__)
+
+
+class CommunicateAsNewUser(PBA):
+    """
+    This PBA creates a new user, and then pings google as that user. This is used for a Zero Trust test of the People
+    pillar. See the relevant telemetry processing to see what findings are created.
+    """
+
+    def __init__(self):
+        super(CommunicateAsNewUser, self).__init__(name=POST_BREACH_COMMUNICATE_AS_NEW_USER)
+
+    def run(self):
+        username = USERNAME + ''.join(random.choice(string.ascii_lowercase) for _ in range(5))
+        if is_windows_os():
+            if not self.try_to_create_user_windows(username, PASSWORD):
+                return  # no point to continue if failed creating the user.
+
+            try:
+                # Importing these only on windows, as they won't exist on linux.
+                import win32con
+                import win32process
+                import win32security
+                # Logon as new user: https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-logonusera
+                new_user_logon_token_handle = win32security.LogonUser(
+                    username,
+                    ".",  # use current domain
+                    PASSWORD,
+                    win32con.LOGON32_LOGON_INTERACTIVE,  # logon type - interactive (normal user)
+                    win32con.LOGON32_PROVIDER_DEFAULT)  # logon provider
+            except Exception as e:
+                PostBreachTelem(
+                    self,
+                    ("Can't logon as {}. Error: {}".format(username, e.message), False)
+                ).send()
+                return  # no point to continue if can't log on.
+
+            # Using os.path is OK, as this is on windows for sure
+            ping_app_path = os.path.join(os.environ["WINDIR"], "system32", "PING.exe")
+            if not os.path.exists(ping_app_path):
+                PostBreachTelem(self, ("{} not found.".format(ping_app_path), False)).send()
+                return  # Can't continue without ping.
+
+            try:
+                # Open process as that user:
+                # https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessasusera
+                commandline = "{} {}".format(ping_app_path, "google.com")
+                _ = win32process.CreateProcessAsUser(
```

Comment:
By not checking return values you are leaking a process and thread handle. Neither will be closed untill monkey finishes execution. Please check return value and close handles.

Reply:
Done